src/services/completions.ts

```diff
@@ -1978,13 +2009,15 @@ namespace ts.Completions {
     interface CompletionEntryDisplayNameForSymbol {
         readonly name: string;
         readonly needsConvertPropertyAccess: boolean;
+        readonly needsConvertAwait: boolean;
     }
     function getCompletionEntryDisplayNameForSymbol(
         symbol: Symbol,
         target: ScriptTarget,
         origin: SymbolOriginInfo | undefined,
         kind: CompletionKind,
     ): CompletionEntryDisplayNameForSymbol | undefined {
+        const needsConvertAwait = !!(origin && originIsPromise(origin));

```

Comment:
I don't think this needs to be passed around since this check can easily be performed as usage sites of `needsConvertAwait`